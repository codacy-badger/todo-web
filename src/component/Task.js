import React from "react";
import "component/Task.sass";
import { faCheckCircle } from "@fortawesome/free-regular-svg-icons";
import { faCheckCircle as SolidFaCheckCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

function Task({ text = "Task", completed = false, onClick = null }) {
  return completed ? (
    <div className="task-component completed" onClick={onClick}>
      <FontAwesomeIcon className="icon" icon={faCheckCircle} />
      {text}
    </div>
  ) : (
    <div className="task-component incomplete" onClick={onClick}>
      <FontAwesomeIcon className="icon" icon={SolidFaCheckCircle} />
      {text}
    </div>
  );
}

export default Task;
