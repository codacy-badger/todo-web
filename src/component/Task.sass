@import "colors.sass"

.task-component
  font-weight: bold
  cursor: pointer
  font-size: 22px

  .icon
    width: 20px
    height: 20px
    margin-right: 10px
    transition: 0.5s ease all

  &.incomplete
    &:hover
      .icon
        opacity: 1

    .icon
      opacity: 0

  &.completed
    font-weight: normal
    color: $text-secondary
