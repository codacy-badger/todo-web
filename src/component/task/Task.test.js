import React from "react";
import renderer from "react-test-renderer";
import Task from "component/task/Task";

const TEST_TASK_COMPLETED = {
  text: "Completed task",
  completed: true
};

const TEST_TASK_UNCOMPLETE = {
  text: "Uncompleted task",
  completed: false
};

describe("Task component", () => {
  test("should render default Task", () => {
    const component = renderer.create(<Task />);
    expect(component).toMatchSnapshot();
  });

  test("should render completed Task", () => {
    const component = renderer.create(
      <Task
        text={TEST_TASK_COMPLETED.text}
        completed={TEST_TASK_COMPLETED.completed}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("should render uncomplete Task", () => {
    const component = renderer.create(
      <Task
        text={TEST_TASK_UNCOMPLETE.text}
        completed={TEST_TASK_UNCOMPLETE.completed}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
