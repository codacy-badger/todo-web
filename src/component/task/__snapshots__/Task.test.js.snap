// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component should render completed Task 1`] = `
<div
  className="task-component completed"
  onClick={null}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-check-circle fa-w-16 icon"
    data-icon="check-circle"
    data-prefix="far"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  Completed task
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-trash-alt fa-w-14 icon-trash"
    data-icon="trash-alt"
    data-prefix="far"
    focusable="false"
    onClick={null}
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`Task component should render default Task 1`] = `
<div
  className="task-component incomplete"
  onClick={null}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-check-circle fa-w-16 icon"
    data-icon="check-circle"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  Task
</div>
`;

exports[`Task component should render uncomplete Task 1`] = `
<div
  className="task-component incomplete"
  onClick={null}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-check-circle fa-w-16 icon"
    data-icon="check-circle"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  Uncompleted task
</div>
`;
